### PR TITLE
Feature/323 install go tools

### LINKED ISSUE #323: Install tool go-licenses automatically
Tool `go-licenses`  is used by `GolangServices.getLicenses()` but requires prior installation with 
```sh
go install github.com/google/go-licenses@v1.2.1
```
The current ticket proposes PK to install the `go-licenses` automatically. 
By that issues of wrong versions can be avoided.

Proposal: 
1. Class `GoProcess` checks if command to be executed is already installed in the place indicated by `GoProcess`, `GOPATH`, "bin", ..
2. if not installed already then install it 

Notes:
* This implementation can also be used in `ci-build.yml`.
* The preferred version can be hard-coded in PK or replaced by latest version.

### PR DESCRIPTION
**Pull request by ckunki**
Fixes #323

### CODE REVIEW

**Review by kaklakariada — CHANGES_REQUESTED** [review]
Code structure looks very good! Only some minor refactoring :)

File: project-keeper/src/main/java/com/exasol/projectkeeper/sources/analyze/golang/GoBinary.java [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    }
+
+    /**
+     * Check if current {@link GoBinary} is already installed and install it if required.
+     *
+     * @param workingDirectory working directory to execute potential installation
+     * @param goProcess        go process to launch potential installation
+     * @return this for fluent programming
+     */
+    public GoBinary install(final Path workingDirectory, final GoProcess goProcess) {
```
[the remark this answers is not in the document]
Comment (ckunki) [reply project-keeper/src/main/java/com/exasol/projectkeeper/sources/analyze/golang/GoBinary.java:51]: 👍

File: project-keeper/src/main/java/com/exasol/projectkeeper/sources/analyze/golang/GoBinary.java [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+    public String command() {
+        final Path path = path();
+        if (Files.exists(path)) {
+            return path.toString();
+        }
+        return this.name;
+    }
+
+    Path path() {
+        return getGoBinPath().resolve("bin").resolve(nameWithSuffix());
```
[the remark this answers is not in the document]
Comment (ckunki) [reply project-keeper/src/main/java/com/exasol/projectkeeper/sources/analyze/golang/GoBinary.java:87]: not used for go binary itself

File: project-keeper/src/test/java/com/exasol/projectkeeper/sources/analyze/golang/GoBinaryTest.java [hunk 3]
```diff
@@ -7,20 +7,24 @@
...
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
 
+@ExtendWith(MockitoExtension.class)
 class GoBinaryTest {
 
     private static final String SEP = File.separator.replace("\\", "\\\\");
-    private static final GoBinary TESTEE = new GoBinary(null, "non-existing-go-binary", null);
+    @Mock
+    private final GoProcess goProcess = mock(GoProcess.class);
```
Comment (kaklakariada) [inline project-keeper/src/test/java/com/exasol/projectkeeper/sources/analyze/golang/GoBinaryTest.java:23]:
```suggestion
    @Mock
    private GoProcess goProcess;
```
Mock should be created automatically

**Review by kaklakariada — CHANGES_REQUESTED** [review]

**Review by kaklakariada — DISMISSED** [review]

**Review by kaklakariada — APPROVED** [review]

**Pull request merged**